daemon/cmd/fqdn.go

```diff
@@ -394,7 +394,8 @@ func (d *Daemon) updateSelectors(ctx context.Context, selectorWithIPsToUpdate ma
 
 // lookupEPByIP returns the endpoint that this IP belongs to
 func (d *Daemon) lookupEPByIP(endpointIP net.IP) (endpoint *endpoint.Endpoint, err error) {
-	e := d.endpointManager.LookupIP(endpointIP)
+	endpointAddr, _ := ippkg.AddrFromIP(endpointIP)
+	e := d.endpointManager.LookupIP(endpointAddr)
```

Comment:
It looks liked `LookupIP` does not check `endpointAddr` to be valid but `endpointIP` could potentially be `nil` (and thus `endpointAddr` invalid) if I read https://github.com/cilium/cilium/blob/f28b8e262a152d046b8c393e6a3448baca050c38/pkg/fqdn/dnsproxy/proxy.go#L791 correctly. So we should probably check the `, ok` value here to be safe:

```suggestion
	endpointAddr, ok := ippkg.AddrFromIP(endpointIP)
	if !ok {
		return nil, fmt.Errorf("invalid IP %s for endpoint lookup", endpointIP)
	}
	e := d.endpointManager.LookupIP(endpointAddr)
```